src/types.ts

```diff
@@ -220,6 +220,61 @@ export type ITick = {
   key?: number;
 };
 
+export class SelectionProxy {
```

Comment:
`types.ts` is intended only for interfaces, not implementations (this way, you can `import type ...`). This class can move to another file.